domain_models/models.py

```diff
@@ -211,3 +212,11 @@ def get_data(self):
         return dict((name, field.get_builtin_type(self))
                     for name, field in
                     six.iteritems(self.__class__.__fields__))
+
+    def set_data(self, data):
+        """Set dictionary data to model.
+
+        :param dict data:
+        """
+        for name, value in six.iteritems(data):
+            self.__class__.__fields__[name].set_builtin_type(self, value)
```

Comment:
I think that here you need to put ``self.__class__.__fields__[name]`` in ``try / except KeyError`` block (like you did it in ``get()``) for making more verbose output when field doesn't exist.

Also, at some point, it could be an option to iterate fields and try to get values for them from ``data``. At this point we will ignore all parasite values in ``data`` (do we want this? - I'm not sure, may be).